src/utils/wrappedCurrency.ts

```diff
@@ -1,9 +1,11 @@
-import { Currency, WETH } from '@kyberswap/ks-sdk-core'
+import { Currency, Token, WETH } from '@kyberswap/ks-sdk-core'
 
+import { ZERO_ADDRESS } from 'constants/index'
 import { NativeCurrencies } from 'constants/tokens'
 
 export function unwrappedToken(token: Currency): Currency {
   if (token.equals(WETH[token.chainId])) return NativeCurrencies[token.chainId]
+  if (token instanceof Token && token.address === ZERO_ADDRESS) return NativeCurrencies[token.chainId]
```

Comment:
how about token.wrapped.address?

Reply:
nice point, but this `token: Currency` could be some fake `object: any` force cast into `Currency` (legacy code), so .wrapped will result in error